packages/graphql_server/src/scraping/twitterScraping.ts

```diff
@@ -0,0 +1,79 @@
+import axios from 'axios'
+import { TwitterPost, TwitterSearchResponse } from '../../types/twitterScraping'
+
+const apiKey = process.env.SCRAPER_API_API_KEY || ''
+const SORT_BY_RETWEETS = 'retweets'
+const SORT_BY_REPLIES = 'replies'
+
+/**
+ * Gets first page of twitter posts for a given hashtag
+ * @param hashtag - the hashtag to search for
+ * @returns An array of the TwitterPosts
+ */
+export async function getPostsForHashtag(hashtag: string): Promise<TwitterPost[]> {
+  let result: TwitterSearchResponse
+  try {
+    const response = await axios.get<TwitterSearchResponse>(
+      'https://api.scraperapi.com/structured/twitter/v2/search',
+      {
+        params: {
+          api_key: apiKey,
+          query: hashtag
+        }
+      }
+    )
+    result = response.data
+    return mapToInternalType(result.tweets)
+  } catch (error) {
+    console.log('Could not fetch twitter search results for: ' + hashtag)
+    return []
+  }
+}
+
+/**
+ * Gets first page of twitter posts for a given hashtag and sorts the tweets by given criterion
+ * @param hashtag - the hashtag to search for
+ * @param sortBy - the sorting criterion
+ * @returns An sorted array of the TwitterPosts
+ */
+export async function getPostsForHashtagSortedBy(
+  hashtag: string,
+  sortBy: string
+): Promise<TwitterPost[]> {
+  try {
+    const tweets = await getPostsForHashtag(hashtag)
+
+    return tweets.sort((a, b) => {
+      if (sortBy === SORT_BY_REPLIES) {
+        return b.replies - a.replies
+      }
+      if (sortBy === SORT_BY_RETWEETS) {
+        return b.retweetCount - a.retweetCount
+      }
+      return 0
+    })
+  } catch (error) {
+    console.log('Could not fetch twitter search results for: ' + hashtag)
+    return []
+  }
+}
+
+/**
+ * Maps an array of the API response from TwitterSearchResponse type to TwitterPost type.
+ * @param tweets - The array of tweets to be mapped.
+ * @returns An array of mapped TwitterPost objects.
+ */
+function mapToInternalType(tweets: TwitterSearchResponse['tweets']): TwitterPost[] {
+  return tweets.map((tweet) => ({
```

Comment:
in general I think we shouldn't rename fields if not necessary
then it would be possible to set default values with Object.assign
but's ok